drivers/network/netio/include/netio_debug.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * COPYRIGHT:   See COPYING in the top level directory
+ * PROJECT:     ReactOS Ancillary Function Driver
+ * FILE:        include/netio_debug.h
+ * PURPOSE:     Debugging support macros
+ * DEFINES:     DBG     - Enable debug output
+ *              NASSERT - Disable assertions
+ */
+
+/* Question to reviewers: does mingw support this #pragma? */
+#pragma once
+
+#define NORMAL_MASK    0x000000FF
+#define SPECIAL_MASK   0xFFFFFF00
+#define MIN_TRACE      0x00000001
+#define MID_TRACE      0x00000002
+#define MAX_TRACE      0x00000003
+
+#define DEBUG_CHECK    0x00000100
+#define DEBUG_IRP      0x00000200
+#define DEBUG_ULTRA    0xFFFFFFFF
+
+#if DBG
+
+extern ULONG DebugTraceLevel;
+
+#ifdef _MSC_VER
+
+#define NETIO_DbgPrint(_t_, _x_) \
+    if ((_t_ > NORMAL_MASK) \
+        ? (DebugTraceLevel & _t_) > NORMAL_MASK \
+        : (DebugTraceLevel & NORMAL_MASK) >= _t_) { \
+        DbgPrint("(%s:%d) ", __FILE__, __LINE__); \
+        DbgPrint _x_ ; \
+    }
+
+#else /* _MSC_VER */
+
+#define NETIO_DbgPrint(_t_, _x_) \
+    if ((_t_ > NORMAL_MASK) \
+        ? (DebugTraceLevel & _t_) > NORMAL_MASK \
+        : (DebugTraceLevel & NORMAL_MASK) >= _t_) { \
+        DbgPrint("(%s:%d)(%s) ", __FILE__, __LINE__, __FUNCTION__); \
+        DbgPrint _x_; \
+    }
+
+#endif /* _MSC_VER */
+
+#ifdef ASSERT
+#undef ASSERT
+#endif
+
+#ifdef NASSERT
+#define ASSERT(x)
+#else /* NASSERT */
+#define ASSERT(x) if (!(x)) { NETIO_DbgPrint(MIN_TRACE, ("Assertion "#x" failed at %s:%d\n", __FILE__, __LINE__)); DbgBreakPoint(); }
+#endif /* NASSERT */
+
+#else /* DBG */
+
+#define NETIO_DbgPrint(_t_, _x_)
+
+#define ASSERTKM(x)
+#ifndef ASSERT
+#define ASSERT(x)
+#endif
+
+#endif /* DBG */
+
+
+#undef assert
+#define assert(x) ASSERT(x)
+
+
+#ifdef _MSC_VER
+
+#define UNIMPLEMENTED \
+    NETIO_DbgPrint(MIN_TRACE, ("The function at %s:%d is unimplemented, \
```

Comment:
Here's an example of the baroque thing I was talking about above: the parenthesized part for the 2nd parameter and the others.